src/MongoDB.Driver.Core/Core/Servers/TopologyDescription.cs

```diff
@@ -0,0 +1,323 @@
+        /// </returns>
+        public bool IsFresherThanServerResponse(TopologyDescription serverResponse) => CompareFreshnessToServerResponse(serverResponse) > 0;
+
+        /// <summary>
+        /// Gets whether or not this TopologyDescription is fresher than <paramref name="serverResponse"/>.
+        /// Comparing topology descriptions freshness does not exhibit the reversal property of
+        /// inequalities e.g. a.IsStalerThanServerResponse(b) (a "&lt;" b) does not imply
+        /// !b.IsStalerThanServerResponse(a) (b "&gt;" a)
+        /// See <seealso cref="CompareFreshnessToServerResponse"/> for more information.
+        /// In the case that this == <paramref name="serverResponse"/>,
+        /// <paramref name="serverResponse"/> will be consider to be fresher.
+        /// </summary>
+        /// <param name="serverResponse">The TopologyDescription received from a server/ </param>
+        /// <returns>
+        /// Wwhether or not this TopologyDescription is fresher than <paramref name="serverResponse"/>.
+        /// </returns>
+        public bool IsFresherThanOrEqualToServerResponse(TopologyDescription serverResponse) => CompareFreshnessToServerResponse(serverResponse) >= 0;
+
+        /// <summary>
+        /// Gets whether or not this TopologyDescription is staler than <paramref name="serverResponse"/>.
+        /// Comparing topology descriptions freshness does not exhibit the reversal property of
+        /// inequalities e.g. a.IsStalerThanServerResponse(b) (a "&lt;" b) does not imply
+        /// !b.IsStalerThanServerResponse(a) (b "&gt;" a).
+        /// See <seealso cref="CompareFreshnessToServerResponse"/> for more information.
+        /// In the case that this == <paramref name="serverResponse"/>,
+        /// <paramref name="serverResponse"/> will be consider to be fresher.
+        /// </summary>
+        /// <param name="serverResponse">The TopologyDescription received from a server/ </param>
+        /// <returns>
+        /// Wwhether or not this TopologyDescription is staler than <paramref name="serverResponse"/>.
+        /// </returns>
+        public bool IsStalerThanServerResponse(TopologyDescription serverResponse) => CompareFreshnessToServerResponse(serverResponse) <= 0;
+
+        /// <inheritdoc/>
+        public override int GetHashCode() => _hashCode;
+
+        /// <inheritdoc/>
+        public override string ToString() => $"{{ ProcessId : {_processId}, Counter : '{_counter}' }}";
+
+        // explicit interface implementations
+#if NET452 || NETSTANDARD2_0
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.AddValue(nameof(_processId), _processId, typeof(ObjectId));
+            info.AddValue(nameof(_counter), _counter, typeof(long));
+        }
+#endif
+    }
+
+    /// <summary>
+    /// Extensions for Nullable TopologyDescription
+    /// </summary>
+    public static class NullableTopologyDescriptionExtensions
```

Comment:
This static class provides extension methods for `Nullable<TopologyDescription>` and implements the second layer of logic regarding comparisons with nulls.

Reply:
Comment moved to renamed file so it'll show in the diff. Resolving